src/scores/continuous/interval_impl.py

```diff
@@ -0,0 +1,226 @@
+"""
+Implementation of quantile interval and interval scores
+"""
+from typing import Optional
+
+import xarray as xr
+
+from scores.functions import apply_weights
+from scores.typing import FlexibleDimensionTypes
+from scores.utils import check_dims, gather_dimensions
+
+
+def quantile_interval_score(  # pylint: disable=R0914
+    fcst_lower_qtile: xr.DataArray,
+    fcst_upper_qtile: xr.DataArray,
+    obs: xr.DataArray,
+    lower_qtile_level: float,
+    upper_qtile_level: float,
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.Dataset:
+    """
+    Calculates the qauntile interval score for interval forecasts. This score penalizes
+    the interval's width and whether the observation value lies outside the interval.
+
+    .. math::
+    \\text{quantile interval score} = (q_{u} - q_{l}) +
+    \\frac{1}{\\alpha_l} \\cdot (q_{l} - y) \\cdot \\mathbb{1}(y < q_{l}) + \\frac{1}{1 - \\alpha_u} \\cdot (q_{u} - y) \\cdot \\mathbb{1}(y < q_{u})
+
+    where
+        - :math: `q_{u}` is the forecast at upper quantile
+        - :math: `q_{l}` is the forecast at lower quantile
+        - :math: `\\alpha_u` is upper quantile level
+        - :math: `\\alpha_l` is lower quantile level
+        - :math: `y` is the observation
+        - :math: `\\mathbb{1}(condition)` is an indicator function that is 1 when the condition is true, and 0 otherwise.
```

Comment:
```suggestion
    .. math::
        \\text{quantile interval score} = (q_{u} - q_{l}) +
        \\frac{1}{\\alpha_l} \\cdot (q_{l} - y) \\cdot \\mathbb{1}(y < q_{l}) + \\frac{1}{1 - \\alpha_u} \\cdot (q_{u} - y) \\cdot \\mathbb{1}(y < q_{u})

    where
        - :math:`q_u` is the forecast at upper quantile
        - :math:`q_{l}` is the forecast at lower quantile
        - :math:`\\alpha_u` is upper quantile level
        - :math:`\\alpha_l` is lower quantile level
        - :math:`y` is the observation
        - :math:`\\mathbb{1}(condition)` is an indicator function that is 1 when the condition is true, and 0 otherwise.
```